src/globalPlug.ts

```diff
@@ -46,62 +56,57 @@ export class GlobalPlug implements Plug {
 
         const logger = this.instance.getLogger();
         const pending: Promise<void>[] = [];
-        const initialized: string[] = [];
 
-        for (const plugin of plugins ?? []) {
-            const pluginName = plugin.getName();
+        for (const [name, options] of Object.entries(plugins ?? {})) {
+            logger.debug(`Initializing plugin "${name}"...`);
 
-            if (initialized.includes(pluginName)) {
-                logger.warn(
-                    `Multiple plugins registered with name "${pluginName}" `
-                    + 'which may cause unexpected errors.',
-                );
-            }
-
-            logger.debug(`Initializing plugin "${pluginName}"...`);
+            const factory = this.pluginFactories[name];
 
-            initialized.push(pluginName);
+            if (factory === undefined) {
+                throw new Error(`Plugin '${name}' is not registered.`)
+            }
```

Comment:
Before the plugins, there was no statement that could throw an error anywhere after the `SdkFacade.init`.
Now that it can happen, what are the consequences of having a Plug initialized 2 times, one with a plugin error and another without?

It leaves an orphan initialized SDK that would not be collected because it has event emitters referencing it and all the plugins instantiated up to the failure may also be left like that.

Should we deal with this or just treat as user misuse and tell them to deploy a correct version?